eudi_wallet_python/satosa/backend.py

```diff
@@ -0,0 +1,157 @@
+import json
+import logging
+import base64
+import satosa.util as util
+
+from six import text_type
+from satosa.context import Context
+from satosa.exception import SATOSAAuthenticationError
+from satosa.response import Response
+from satosa.backends.base import BackendModule
+
+logger = logging.getLogger(__name__)
+
+class OpenIDVP4SAMLBackend(BackendModule):
+    """
+    A saml2 backend module (acting as a OpenIDVP4 SP).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.qrCode_url = self.config['qrCode_enpoint']
+        self.redirect_url = self.config['redirect_endpoint']
+        self.request_url = self.config['request_enpoint']
+
+    def register_endpoints(self):
+        """
+        Creates a list of all the endpoints this backend module needs to listen to. In this case
+        it's the authentication response from the underlying OP that is redirected from the OP to
+        the proxy.
+        :rtype: Sequence[(str, Callable[[satosa.context.Context], satosa.response.Response]]
+        :return: A list that can be used to map the request to SATOSA to this endpoint.
+        """
+        url_map = []
+        url_map.append((f"^{self.qrCode_url.lstrip('/')}$", self.qrCode_endpoint))
+        url_map.append((f"^{self.redirect_url.lstrip('/')}$", self.redirect_endpoint))
+        url_map.append((f"^{self.request_url.lstrip('/')}$", self.request_enpoint))
+        return url_map
+    
+    def qrCode_endpoint(self, context, *args):
```

Comment:
Please

pre_request_endpoint

It should detect the user agente and offers a QR code in a HTML Page or a redirect with a deeplink

Reply:
Should be resolved in the commit 03c6a0bf559097d76770bfcb61a36613718be6bc